src/server/game/Spells/Spell.cpp

```diff
@@ -2322,7 +2322,8 @@ void Spell::TargetInfo::PreprocessTarget(Spell* spell)
     if (_spellHitTarget)
     {
         // if target is flagged for pvp also flag caster if a player
-        if (unit->IsPvP() && spell->m_caster->GetTypeId() == TYPEID_PLAYER)
+        // but respect current pvp rules (buffing/healing npcs flagged for pvp only flags you if they are in combat)
```

Comment:
```suggestion
        // but respect current pvp rules (buffing/healing npcs flagged for PVP), where player is flagged only if npc is in combat
```
That line looked strange.